GameDesign/Assets/Scripts/MonoBehaviours/Movements/DespawnerWaypoint.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DespawnerWaypoint : Waypoint
+{
+    [Header("Despawner properties")]
+    [SerializeField] float distanceToDespawn;
+
+    void OnDrawGizmos()
+    {
+        Gizmos.color = sphereColor;
+        Gizmos.DrawWireSphere(transform.position, distanceToDespawn);
+        Gizmos.color = connectingColor;
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    public override void waypointReached(NPCMovementManager npcMover)
+    {
+        if(Vector3.Distance(transform.position, npcMover.transform.position) < distanceToDespawn)
+        {
+            Destroy(npcMover.gameObject);
+        }

```

Comment:
in theory this check is redundant because the npcMover should do it itself and only then call the check point treached. since its logical arrival rather than a hitbox check.

Reply:
Is not redundant because you added the TTL, so there's the chance that it didn't arrived to the waypoint